packages/quip-cli/src/commands/login.ts

```diff
@@ -66,11 +67,16 @@ export const login = async ({
     hostname = DEFAULT_HOSTNAME,
     port = DEFAULT_PORT,
     config = defaultConfigPath(),
+    displayTokenOnly = false,
+    // This is added for test purpose, otherwise the mock will have problem to fetch data from stdout.
+    log = console.log
```

Comment:
Does `println` not work?

Reply:
It does not. If `printIn` is used, unit test will receive empty string from stdout.

FYI: `println` works in my local machine when I ran the unit test locally, but it would fail here: https://github.com/quip/quip-apps/runs/4175169643?check_suite_focus=true

Reply:
what about `this.log`

Reply:
`this.log` comes from the abstract class `Command`. And it is not available to `login` method, which is just a static method. https://github.com/quip/quip-apps/blob/44b8000d9c0d5e830136a883435a81242842c463/packages/quip-cli/src/commands/login.ts#L65

`this.log` is only available within `Login` class. https://github.com/quip/quip-apps/blob/44b8000d9c0d5e830136a883435a81242842c463/packages/quip-cli/src/commands/login.ts#L135

Reply:
from https://github.com/quip/quip-apps/blob/44b8000d9c0d5e830136a883435a81242842c463/packages/quip-cli/src/commands/login.ts#L223-L224, `this.log` is passed to `login` method.

Reply:
`login` is used by other quip-cli files so i don't want to put parameters bespoke to the login command. Would it be better if you refactor the code into something like this.
```
        try {
            // a skip write parameter skips writing access token in login function, instead the function should return the token.
            accessToken = accessToken || await login({ site, hostname, port, config, displayTokenOnly });
            if (displayTokenOnly) {
                this.log(accessToken)
            } else {
                await writeSiteConfig(config, site, { accessToken });
                this.log("Successfully logged in.");
            }
        } catch (e) {
            this.error(e);
        }
```

Reply:
This is doable, let me update it soon

Reply:
BTW: original `login` will return Promise<void>, while this change will make it return Promise<string>.

Reply:
Thank you @FlikHu , for your helpful sample code. 
I slightly modified it to save an extra `writeSiteConfig` call for some cases.